unleash-yggdrasil/src/strategy_parsing.rs

```diff
@@ -302,14 +302,15 @@ fn rollout_constraint(mut node: Pairs<Rule>) -> RuleFragment {
                 }
                 custom_stickiness
             }
-            None => context.user_id.clone().or_else(||context.session_id.clone()),
+            None => context
+                .user_id
+                .clone()
+                .or_else(|| context.session_id.clone()),
```

Comment:
Isn't this `context.user_id.or(context.session_id)`

Reply:
It's an Option<&T> at this point, so it needs to be `context.user_id.clone().or(context.session_id.clone())`.

That is better though, nice suggestion, it'll come up with the test

Reply:
Wait, no, clippy doesn't like it, fair comment from clippy - the `or` may not be evaluated when you think it is